lib/engine/step/buy_sell_par_shares.rb

```diff
@@ -188,6 +189,17 @@ def can_ipo_any?(entity)
         !bought? && @game.corporations.any? { |c| c.can_par?(entity) && can_buy?(entity, c.shares.first&.to_bundle) }
       end
 
+      def purchasable_companies_from_others(entity)
+        @game.purchasable_companies.reject { |c| c.owner == entity }
+      end
+
+      def can_buy_any_company?(entity)
```

Comment:
i think this is not necessary you can just put this all in purchasebl_companies

Reply:
Done